lxd/backup.go

```diff
@@ -1,33 +1,30 @@
 package main
 
 import (
-	"archive/tar"
 	"fmt"
-	"io"
+	"io/ioutil"
 	"os"
-	"os/exec"
 	"path/filepath"
-	"strings"
 	"time"
 
 	"context"
 
 	"gopkg.in/yaml.v2"
 
+	"github.com/lxc/lxd/lxd/backup"
 	"github.com/lxc/lxd/lxd/cluster"
 	"github.com/lxc/lxd/lxd/db"
 	"github.com/lxc/lxd/lxd/operations"
 	"github.com/lxc/lxd/lxd/state"
 	"github.com/lxc/lxd/lxd/task"
 	"github.com/lxc/lxd/shared"
-	"github.com/lxc/lxd/shared/api"
 	log "github.com/lxc/lxd/shared/log15"
 	"github.com/lxc/lxd/shared/logger"
 	"github.com/pkg/errors"
 )
 
 // Load a backup from the database
-func backupLoadByName(s *state.State, project, name string) (*backup, error) {
+func backupLoadByName(s *state.State, project, name string) (*backup.Backup, error) {
```

Comment:
@monstermunchkin one of the aims of moving the backup code to a separate package is so that it unblocks us moving the containerLXC struct out of the `main` package and its own package. This function `backupLoadByName` is called by containerLXC, so if this remains it `main` it will prevent us from moving containerLXC. It would be great to have this function be in the `backup` package too.

Reply:
OK. Then, I need to figure out a nice way of moving this since it depends on `instanceLoadById` which again is in `main`.

Reply:
@monstermunchkin hrm, interesting, this is similar to the issue I had with moving containerLXC here https://github.com/lxc/lxd/pull/6237 it escalates quickly :)

Take a look at https://github.com/lxc/lxd/blob/master/lxd/device/device_utils_instance.go#L15 and https://github.com/lxc/lxd/blob/master/lxd/container.go#L57 for how I overcame a similar issue when creating the `device` package, by creating a package level variable for the function I needed and then linking it via the init() function.

Reply:
@monstermunchkin you might also need to create a backup-local `Instance` interface just with the functions needed for use when storing it inside the `backup` struct.